### PR TITLE
Add a migration section

### PR DESCRIPTION
**Pull request by Sreekala-Gopakumar**
https://github.com/eclipse-openj9/openj9-docs/issues/935

Added migrating section (from Java 8 to Java 11 and from Java 11 to Java 17).

Signed-off-by: Sreekala Gopakumar sreekala.gopakumar@ibm.com

### CODE REVIEW

File: docs/migrating11to17.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+-->
+
+
+# Migrating from Java 11 to Java 17
+
+Support for OpenJDK 17 was added in OpenJ9 version 0.29.1.
+
+The following new OpenJ9 changes apply when OpenJ9 is built with Java SE 17 class libraries. This information exists elsewhere in the documentation but is summarized here for convenience:
+
+- Support for JDK enhancement proposals (JEPs) as described in [What's new in version 0.29.1](version0.29.1.md).
```
Comment (pshipton) [inline docs/migrating11to17.md:32]: This should also include the JEPs added in Java 12 - 16.

File: docs/migrating8to11.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+* SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH
+* Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+-->
+
+
+# Migrating from Java 8 to Java 11
+
+Support for OpenJDK 11 was added in OpenJ9 version 0.10.0.
+
+The following new OpenJ9 features and notable changes apply when OpenJ9 is built with Java SE 11 class libraries. This information exists elsewhere in the documentation but is summarized here for convenience:
```
Comment (pshipton) [inline docs/migrating8to11.md:30]: The Java 17 migration refers to a JEP list. Ideally we should also have a JEP list for Java 11 migration, that includes applicable JEPs from Java 9 - 11.

File: docs/migrating11to17.md (hunk 1 above)
Comment (Sreekala-Gopakumar) [reply docs/migrating11to17.md:32]: @pshipton - We have JEP list only for Java 15, 16 and 17, not any earlier releases. We can add the JEP list to the What's new topic of the earlier releases, if you could provide the same. 

We have a JEP list for Java 9, 10, and 11 in https://www.ibm.com/docs/en/semeru-runtime-ce-z/11?topic=new-first-release. Can this list be used here?

Comment (pshipton) [reply docs/migrating11to17.md:32]: > We have a JEP list for Java 9, 10, and 11. Can this list be used here?

Sure.

Comment (pshipton) [reply docs/migrating11to17.md:32]: Raw list https://openjdk.org/projects/jdk/15

339: 	Edwards-Curve Digital Signature Algorithm (EdDSA) - OpenJDK
360: 	Sealed Classes (Preview)
371: 	Hidden Classes
372: 	Remove the Nashorn JavaScript Engine - OpenJDK
373: 	Reimplement the Legacy DatagramSocket API - OpenJDK
375: 	Pattern Matching for instanceof (Second Preview) - OpenJDK
378: 	Text Blocks - OpenJDK
383: 	Foreign-Memory Access API (Second Incubator)
384: 	Records (Second Preview)
385: 	Deprecate RMI Activation for Removal - OpenJDK

Comment (pshipton) [reply docs/migrating11to17.md:32]: We don't really need to distinguish between what required changes to OpenJ9 and what did not.

**Comment by Sreekala-Gopakumar** [comment]
Jenkins doc stage

**Comment by Sreekala-Gopakumar** [comment]
@pshipton, I have made the changes based on your review feedback. Could you please let me know if the changes are fine? Thanks!

File: docs/version0.9.md [hunk 3]
```diff
@@ -135,6 +137,93 @@ For performance and compatibility with the new String object layout, the OpenJ9
...
+- [JEP 299](https://openjdk.org/jeps/299): Reorganize Documentation
+
+You can find the full list of features for JDK 9 at the [OpenJDK project](http://openjdk.java.net/projects/jdk9/). Any remaining features that are listed do not apply to OpenJ9.
+
+### ![Start of content that applies to Java 10 plus](cr/java10plus.png) New JDK 10 features
+
+The following features are supported by OpenJ9:
+
+- [JEP 286](https://openjdk.org/jeps/286): Local-Variable Type Inference
+- [JEP 296](https://openjdk.org/jeps/296): Consolidate the JDK Forest into a Single Repository
```
Comment (pshipton) [inline docs/version0.9.md:219]: This one shouldn't be listed, it has nothing to do with the runtime, but how the OpenJDK source code is organized.

File: docs/version0.10.md [hunk 4]
```diff
@@ -62,6 +63,26 @@ For compatibility, the following OpenJDK HotSpot options are now supported by Op
...
+- [JEP 181](https://openjdk.org/jeps/181): Nest-Based Access Control
+- [JEP 309](https://openjdk.org/jeps/309): Dynamic Class-File Constants
+- [JEP 318](https://openjdk.org/jeps/318): Epsilon: A No-Op Garbage Collector
+- [JEP 320](https://openjdk.org/jeps/320): Remove the Java EE and CORBA Modules
+- [JEP 321](https://openjdk.org/jeps/321): HTTP Client (Standard)
+- [JEP 323](https://openjdk.org/jeps/323): Local-Variable Syntax for Lambda Parameters
+- [JEP 324](https://openjdk.org/jeps/324): Key Agreement with Curve25519 and Curve448
+- [JEP 329](https://openjdk.org/jeps/329): ChaCha20 and Poly1305 Cryptographic Algorithms
+- [JEP 330](https://openjdk.org/jeps/330): Launch Single-File Source-Code Programs
+- [JEP 331](https://openjdk.org/jeps/331): Low-Overhead Heap Profiling
```
Comment (pshipton) [inline docs/version0.10.md:79]: This should have a note about the restrictions mentions earlier in this topic.

Comment (pshipton) [reply docs/version0.10.md:79]: Sorry, the restrictions you have mentioned in migrating8to11.md

File: docs/version0.19.md [hunk 5]
```diff
@@ -69,6 +70,24 @@ The `jpackage` utility is described in JEP 343 as a tool that "packages a Java a
 
 Linux x86 64-bit, Linux on POWER&reg; LE 64-bit, and Linux on IBM Z&reg; 64-bit have all moved to the gcc 7.5 compiler. See [Supported environments](openj9_support.md).
 
+### ![Start of content that applies to Java 14 plus](cr/java14plus.png) New JDK 14 features
+
+The following features are supported by OpenJ9:
+
+- [JEP 343](https://openjdk.org/jeps/343): Packaging Tool (Incubator)
```
Comment (pshipton) [inline docs/version0.19.md:77]: This should refer to the restrictions, https://github.com/eclipse-openj9/openj9-docs/issues/474

File: docs/version0.19.md [hunk 6]
```diff
@@ -69,6 +70,24 @@ The `jpackage` utility is described in JEP 343 as a tool that "packages a Java a
...
+- [JEP 352](https://openjdk.org/jeps/352): Non-Volatile Mapped Byte Buffers
+- [JEP 358](https://openjdk.org/jeps/358): Helpful NullPointerExceptions
+- [JEP 359](https://openjdk.org/jeps/359): Records (Preview)
+- [JEP 370](https://openjdk.org/jeps/370): Foreign-Memory Access API (Incubator)
```
Comment (pshipton) [inline docs/version0.19.md:81]: We should remove this, I'm not sure we actually supported this incubator in jdk 14.

File: docs/version0.10.md (hunk 4 above)
Comment (Sreekala-Gopakumar) [reply docs/version0.10.md:79]: While incorporating this change, we noticed that JEP 331 is discussed in What's new in Version0.10.0 topic as well as What's new Version0.15.1 topic. Is it supposed to be discussed in both? If not, then from which topic should it be deleted?

Thanks!

Comment (pshipton) [reply docs/version0.10.md:79]: JEP 331 support wasn't available in the 0.10 release, but was added later in 0.15.1

**Comment by Sreekala-Gopakumar** [comment]
Jenkins doc stage

**Comment by Sreekala-Gopakumar** [comment]
@pshipton - I have incorporated all the review comments. Are the changes fine now?

Thanks!

**Comment by pshipton** [comment]
lgtm. I suggest squashing the commits before this is merged.

**Comment by Sreekala-Gopakumar** [comment]
@pshipton - As advised I have squashed the commits. Do let me know if there is anything else to be done.

Thanks!

**Comment by Sreekala-Gopakumar** [comment]
@pshipton - I have reworded the commit message. Please check and confirm that the changes are fine now. Thanks!

**Review by pshipton — APPROVED** [review]

**Pull request merged**